### DISCUSSION TITLE
Platform property - kompose convert to k8s

### DISCUSSION THREAD

**Discussion by pfk-beta** [cookiecutter/cookiecutter-django#3721] [Q&A] [answered]
Hello,

First of all, thanks for great software cookiecutter-django! I'm trying to deploy project created with ccdj to k8s cluster. At the begin I want to try tool for conversion, which is kompose. But when I try to use latest version, I'm getting following error:
```
FATA services.celerybeat Additional property platform is not allowed 
```

My question what is purpose of using platform property in docker-compose?

**Comment by Andrew-Chen-Wang** [comment, marked as answer]
This is the commit that led to the platform property: https://github.com/cookiecutter/cookiecutter-django/commit/d8fdd3ddceb19958fa24849ad226fb8d056d3560 or PR #3562

**Reply by pfk-beta** [reply]
Almost wonderfull answer and I'm dummy that I didn't check git blame. This means that this property is only for cornercase when I'm using Macbook.

**Reply by foarsitter** [reply]
Does kompose work as expected when you remove the platform property?

**Reply by pfk-beta** [reply]
@foarsitter yes, and docker-compose as well.

**Reply by pfk-beta** [reply]
```
/usr/local/bin/kompose convert                    
WARN Service "celerybeat" won't be created because 'ports' is not specified 
WARN Service "celeryworker" won't be created because 'ports' is not specified 
WARN Service "django" won't be created because 'ports' is not specified 
WARN Service "flower" won't be created because 'ports' is not specified 
WARN Service "postgres" won't be created because 'ports' is not specified 
WARN Service "redis" won't be created because 'ports' is not specified 
INFO Kubernetes file "traefik-service.yaml" created 
INFO Kubernetes file "celerybeat-deployment.yaml" created 
INFO Kubernetes file "envs--production--django-configmap.yaml" created 
INFO Kubernetes file "envs--production--postgres-configmap.yaml" created 
INFO Kubernetes file "celeryworker-deployment.yaml" created 
INFO Kubernetes file "django-deployment.yaml" created 
INFO Kubernetes file "flower-deployment.yaml" created 
INFO Kubernetes file "postgres-deployment.yaml" created 
INFO Kubernetes file "production-postgres-data-persistentvolumeclaim.yaml" created 
INFO Kubernetes file "production-postgres-data-backups-persistentvolumeclaim.yaml" created 
INFO Kubernetes file "redis-deployment.yaml" created 
INFO Kubernetes file "traefik-deployment.yaml" created 
INFO Kubernetes file "production-traefik-persistentvolumeclaim.yaml" created
```

**Comment by foarsitter** [comment]
I spend to much time learning k8s without knowing the basics properly. The learning-curve of k8s is very steep and if you are not familiar with docker/containerization you should get familiar with this first.

In example start with docker-compose first en read about the platform option: https://docs.docker.com/compose/compose-file/#platform

**Reply by pfk-beta** [reply]
@foarsitter I'm pretty sure you spend a lot of time learning many things in IT, like me. You cannot learn everything, because nobody can have infinite time, but always you can ask questions like mine to the community like here. I agree I'm not expert in k8s, but I have sufficient knowledge in it, and in docker to develop and maintain webapps. I was just asking what was the purpose of setting this property, and it is cornercase which didn't touch my setup. Have a nice weekend
